build/template-run-mi-e2e-azurearc.yaml

```diff
@@ -0,0 +1,40 @@
+parameters:
+  BuildConfiguration: 'Release'
+  TargetFramework: 'net8.0'
+
+steps:
+# Restore 
+- task: DotNetCoreCLI@2
+  displayName: Restore E2E project
+  inputs:
+    command: restore
+    projects: tests/Microsoft.Identity.Test.E2E/Microsoft.Identity.Test.E2E.MSI.csproj
+    restoreArguments: >
+      /p:RestoreTargetFrameworks=net8.0
+      /p:INCLUDE_MOBILE_AND_LEGACY_TFM=
```

Comment:
nit: not needed?

Reply:
this is how we do not build MAUI and only build net 8 transitively.